docs/content/python/guides/rbac.md

```diff
@@ -0,0 +1,326 @@
+
+  @dataclass(frozen=True)
+  class Repository:
+      name: str
+      organization: Organization
+
+  @dataclass(frozen=True)
+  class Role:
+      name: str
+      resource: Union[Repository, Organization]
+
+  @dataclass
+  class User:
+      name: str
+      roles: Set[Role]
+
+      def assign_role_for_resource(self, name, resource):
+          self.roles.add(Role(name, resource))
+  ```
+
+  If, for example, repository roles and organization roles were stored
+  separately instead of in a heterogeneous set, we might define a pair of
+  `has_role()` rules, one for each role type:
+
+  ```polar
+  has_role(user: User, name: String, repository: Repository) if
+    role in user.repository_roles and
+    role matches { name: name, resource: resource };
+
+  has_role(user: User, name: String, organization: Organization) if
+    role in user.organization_roles and
+    role matches { name: name, resource: resource };
+  ```
+
+{{% /callout %}}
+
+Our `has_role()` rule can check role assignments on repositories and
+organizations, but so far we've only talked about repository roles. Let's
+change that and see how Oso can leverage parent-child relationships like the
+one between `Repository` and `Organization` to grant a role on a child resource
+to a role on the parent.
+
+<!-- TODO(gj): better heading -->
+## Grant a role on a child resource to a role on the child's parent
+
+If you've used ~GitHub~ *GitClub* before, you know that having a role on an
+organization grants certain roles and permissions on that organization's
+repositories. For example, a user is granted the `"maintainer"` role on a
+repository if they're assigned the `"owner"` role on the repository's parent
+organization. This is how you write that rule with Oso:
+
+{{< code file="rbac.polar" >}}
+resource Repository {
+  # ...
+
+  relations = { parent: Organization };
+
+  "maintainer" if "owner" on "parent";
+}
+
+has_relation(organization: Organization, "parent", repository: Repository) if
+  organization = repository.organization;
+{{< /code >}}
+
+First, we declare that the `"parent"` relation for a `Repository` is an
+`Organization`:
+
+```polar
+relations = { parent: Organization };
+```
+
+This is a dictionary where each key is the name of the relation and each value
+is the relation's type.
+
+Next, we write a `has_relation()` rule that tells Oso how to check if an
+organization has the `"parent"` relation with a repository:
+
+```polar
+has_relation(organization: Organization, "parent", repository: Repository) if
+  organization = repository.organization;
+```
+
+<!-- TODO(gj): better phrasing for the next sentence -->
+In this case, an organization is the `"parent"` of a repository if the
+repository's `organization` field points to it.
+
+Finally, we add a shorthand rule that involves the `"maintainer"` repository
+role, the `"owner"` organization role, and the `"parent"` relation between the
+two resource types:
+
+```polar
+"maintainer" if "owner" on "parent";
```

Comment:
maybe include the resource block again for context here, a bit has passed since the code snippet above

Reply:
6652fa00